wlroots-sys/build.rs

```diff
@@ -103,13 +129,399 @@ fn main() {
+    }
+
+    if check_version(
+        "cmake".to_string(),
+        "--version".to_string(),
+        "3.0".to_string(),
+        true
+    )
+    {
+        println!("cmake found");
+    } else {
+        println!("WRONG version of cmake or not installed on system.");
+
+        println!("\nInstallation instructions, install with packet manager or if available");
+        println!("STATIC installation cannot rely on Cargo cmake ");
+        exit(2);
+    }
+
+    if check_version(
+        "clang".to_string(),
+        "--version".to_string(),
+        "6.0".to_string(),
+        true
+    )
+    {
+        println!("clang found");
+    } else {
+        println!("WRONG version of clang or not installed on system.");
+
+        println!("\nInstallation instructions, install with packet manager");
+        println!("STATIC installation cannot rely on Cargo clang ");
+        exit(2);
+    }
+
+    if check_version(
+        "pip3".to_string(),
+        "--version".to_string(),
+        "9.0".to_string(),
+        true
+    )
+    {
+        println!("pip3 found");
+    } else {
+        println!("WRONG version of pip3 or not installed on system.");
+
+        println!("\nInstallation instructions, install with packet manager");
+        exit(2);
+    }
+
+    if check_version(
+        "xml2-config".to_string(),
+        "--version".to_string(),
+        "0.0".to_string(),
+        true
+    )
+    {
+        println!("libxml2-dev found");
+    } else {
+        println!("WRONG version of libxml2-dev or not installed on system.");
+
+        println!("\nInstallation instructions, install with packet manager");
+        exit(2);
+    }
+
+    if check_version(
+        "dot".to_string(),
+        "-V".to_string(),
+        "0.0".to_string(),
+        true
+    )
+    {
+        println!("graphviz found");
+    } else {
+        println!("WRONG version of graphviz or not installed on system.");
+
+        println!("\nInstallation instructions, install with packet manager");
+        exit(2);
+    }
+
+
+    if check_version(
+        "wayland-scanner".to_string(),
+        "--version".to_string(),
+        "0.0".to_string(),
+        false
+    )
+    {
+        println!("libwayland-bin found");
+    } else {
+        println!("WRONG version of libwayland-bin or not installed on system.");
+
+        println!("\nInstallation instructions, install with packet manager");
+        exit(2);
+    }
+
+    println!("other dependencies needed: ");
+    println!("libxml2: ");
+    println!("libwayland-bin: ");
+    println!("wayland-protocols ");
+    println!("libffi-dev: ");
+    println!("expat ");
+    println!("graphviz: ");
+    println!("doxygen: ");
+    println!("llvm ");
+    println!("libwayland-bin: ")
+
+}
+
+/// Checks if a specific package is installed in system PATH or pkg-config
+/// if no min_version is needed use "0" as arg.
+fn check_version(command: String, arg: String, min_version: String, first_check: bool) -> bool {
```

Comment:
If check_version takes &str for the arguments instead String you can write check_version("pkg-config", "--version", "0") for example.